src/cli.rs

```diff
@@ -286,10 +302,7 @@ async fn process_line(readline: Result<String, ReadlineError>, ctx: &mut Context
 
             LineResult::Success(line.to_string())
         }
-        Err(ReadlineError::Interrupted) => {
-            println!("CTRL-C");
-            LineResult::Break
-        }
+        Err(ReadlineError::Interrupted) => LineResult::Error(ShellError::string("CTRL-C")),
```

Comment:
This is still hit when a subprocess isn't running.

Reply:
Can you break out of the subprocess with ctrl-C and land back into nu?

Reply:
I'm not sure I'm following. The `if ctrl_c.load(Ordering::SeqCst) {` on line 94 handles breaking out of a subprocess.

Reply:
So from inside running nu, if you start up an subprocess (say, running a long-running or one that waits for user input), and then while that app is running you hit ctrl-c, do you break out of that app and back to nu?

Reply:
It's okay if the ctrl-c trap gets called even when a subprocess isn't running. We'll just capture it and decide what to do with it

Reply:
> So from inside running nu, if you start up an subprocess (say, running a long-running or one that waits for user input), and then while that app is running you hit ctrl-c, do you break out of that app and back to nu?

Yep, it hit's the trap and is handled on line 94.

If there is no subprocess running it hits this line.

Reply:
Sounds good.  Thanks for the PR!